### DISCUSSION TITLE
Help Accessing Secret Data in NetBox Shell

### DISCUSSION THREAD

**Discussion by paulcarre07** [Onemind-Services-LLC/netbox-secrets#165] [Q&A]
Hello,

I'm new to this plugin, and this is my first time using it.

I’ve created a secret in NetBox linked to a device, and now I’m trying to access the actual secret data via the NetBox shell. So far, I’ve been able to retrieve the secret object using this command:
```
device.secrets.all()[0]
``` 
However, I’m not sure how to access the secret data itself, the plain data. I tried the following command:
```
device.secrets.all()[0].secret
``` 
But it didn’t work as expected.

Could anyone guide me on how to correctly retrieve the secret data? I’d really appreciate the help.

Thanks in advance!

**Comment by llamafilm** [comment]
```python
user_key = UserKey.objects.first()
master_key = user_key.get_master_key(PRIVATE_KEY_PEM)
secret = Secret.objects.first()
secret.decrypt(master_key)
print(secret.plaintext)
```
